main.js

```diff
@@ -86,6 +87,31 @@ function initWindow(window) {
   // get the XUL window that corresponds to this high-level window
   let win = viewFor(window);
 
+  win.addEventListener('TabOpen', win, false);
+  win.addEventListener('TabClose', win, false);
+  win.addEventListener('TabPinned', win, false);
+  win.addEventListener('TabUnpinned', win, false);
+
+  win.handleEvent = function (aEvent) {
```

Comment:
I'm a little worried about someone else overriding the `handleEvent` method on the `win`.  Could we create a new object with a `handleEvent`, and pass that to `win.addEventListener` instead?  (We could even add it to the VerticalTabs instance, if you wanted to make sure it was available everywhere…)